### PR TITLE
Enable subcell for Newtonian Euler

### PR DESCRIPTION
**Pull request by kidder**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp
Comment (nilsdeppe) [inline src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp]: I think we've been trying to spell out `neighbor` instead of using `nhbr` (my fault, sorry)

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM, just one request you can squash immediately. Thanks for doing this :)

**Review by nilsdeppe — DISMISSED** [review]

**Comment by kidder** [comment]
@yoonso0-0 can you provide a second review for this?

**Comment by nilsdeppe** [comment]
@yoonso0-0 can you please review this when you get a chance? Thanks!

@kidder it'll need a rebase

File: src/Evolution/Systems/NewtonianEuler/Subcell/ComputeFluxes.hpp
Comment (yoonso0-0) [inline src/Evolution/Systems/NewtonianEuler/Subcell/ComputeFluxes.hpp:15]: include for `size_t`

File: src/Evolution/Systems/NewtonianEuler/Subcell/ComputeFluxes.hpp
Comment (yoonso0-0) [inline src/Evolution/Systems/NewtonianEuler/Subcell/ComputeFluxes.hpp:26]: This description goes better with the `detail::compute_fluxes_impl()` function above

File: src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp
Comment (yoonso0-0) [inline src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp:66]: For ScalarAdvection system this struct and function are templated as `NeighborPackagedData<Dim>::apply<DbTagsList>`. Not sure which way might be clearer, but I think we need to keep it one way. Any suggestions?

File: src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp
Comment (yoonso0-0) [inline src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp]: default construct this and the line above?

File: src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp
Comment (yoonso0-0) [inline src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp:192]: Q. What is this line for?

File: tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_NeighborPackagedData.cpp
Comment (yoonso0-0) [inline tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_NeighborPackagedData.cpp:245]: include `<memory>`

File: src/Evolution/Systems/NewtonianEuler/Subcell/TimeDerivative.hpp
Comment (yoonso0-0) [inline src/Evolution/Systems/NewtonianEuler/Subcell/TimeDerivative.hpp:55]: Same issue as above (how to template)

**Comment by yoonso0-0** [comment]
@kidder @nilsdeppe I'm currently reviewing this, but I'm not sure if [this commit](https://github.com/sxs-collaboration/spectre/pull/3648/commits/72767107769ab14275e71111b0acf5ef4538bee0) would require another reviewer since I don't know well about coordinate tags.

File: tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_TimeDerivative.cpp
Comment (yoonso0-0) [inline tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_TimeDerivative.cpp]: Not used

File: tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_TimeDerivative.cpp
Comment (yoonso0-0) [inline tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_TimeDerivative.cpp]: Not used

File: tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_TimeDerivative.cpp
Comment (yoonso0-0) [inline tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_TimeDerivative.cpp]: [nitpick] rename to `SmoothFlowMetaVars` for a bit of clarity (since there is `LaneEmdenStarMetaVars` below)

File: src/Evolution/DgSubcell/Actions/Initialize.hpp
Comment (nilsvu) [inline src/Evolution/DgSubcell/Actions/Initialize.hpp]: typo `Coordinates`

File: tests/Unit/Evolution/DgSubcell/Actions/Test_Initialize.cpp
Comment (nilsvu) [inline tests/Unit/Evolution/DgSubcell/Actions/Test_Initialize.cpp:92]: Can you retrieve and check the computed inertial coordinates in this test?

**Review by nilsvu — COMMENTED** [review]
I looked at the one commit that @yoonso0-0 pointed to

File: tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_TimeDerivative.cpp
Comment (yoonso0-0) [inline tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_TimeDerivative.cpp]: Add test for 3D stationary smooth flow?

**Review by yoonso0-0 — COMMENTED** [review]
Please squash :)

File: src/Evolution/Systems/NewtonianEuler/Subcell/ComputeFluxes.hpp
Comment (kidder) [reply src/Evolution/Systems/NewtonianEuler/Subcell/ComputeFluxes.hpp:26]: yes, but documentation doesn't render for code in a detail namespace

File: src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp
Comment (kidder) [reply src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp:192]: it prevents a compiler warning that `dg_mesh` is unused

File: src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp
Comment (kidder) [reply src/Evolution/Systems/NewtonianEuler/Subcell/NeighborPackagedData.hpp:66]: As the only thing in the struct is the apply function, there is no need to template the struct.  Then both template parameters can be deduced from the function call, and don't need to be specified there.

File: src/Evolution/Systems/NewtonianEuler/Subcell/TimeDerivative.hpp
Comment (kidder) [reply src/Evolution/Systems/NewtonianEuler/Subcell/TimeDerivative.hpp:55]: same answer as above

**Comment by kidder** [comment]
@yoonso0-0 @nilsdeppe @nilsleiffischer rebased and squashed

**Review by nilsdeppe — APPROVED** [review]

**Review by yoonso0-0 — APPROVED** [review]

**Pull request merged**
